src/OpenTelemetry.Contrib.Instrumentation.Wcf/README.md

```diff
@@ -0,0 +1,133 @@
+# WCF Instrumentation for OpenTelemetry .NET
+
+Instruments WCF clients and/or services using implementations of
+`IClientMessageInspector` and `IDispatchMessageInspector` respectively.
+
+## Installation
+
+Add the OpenTelemetry.Contrib.Instrumentation.Wcf package via NuGet.
+
+## OpenTelemetry Configuration
+
+Use the `AddWcfInstrumentation` extension method to add WCF instrumentation to
+an OpenTelemetry TracerProvider:
+
+```csharp
+using var openTelemetry = Sdk.CreateTracerProviderBuilder()
+    .AddWcfInstrumentation()
+    .Build();
+```
+
+## WCF Client Configuration (.NET Framework)
+
+Add the `IClientMessageInspector` instrumentation via a behavior extension on
+the clients you want to instrument:
+
+```xml
+<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+  <system.serviceModel>
+    <extensions>
+      <behaviorExtensions>
+        <add name="telemetryExtension" type="OpenTelemetry.Contrib.Instrumentation.Wcf.TelemetryBehaviourExtensionElement, OpenTelemetry.Contrib.Instrumentation.Wcf"   />
+      </behaviorExtensions>
+    </extensions>
+    <behaviors>
+      <endpointBehaviors>
+        <behavior name="telemetry">
+          <telemetryExtension />
+        </behavior>
+      </endpointBehaviors>
+    </behaviors>
+    <bindings>
+      <netTcpBinding>
+        <binding name="netTCPConfig">
+          <security mode="None" />
+        </binding>
+      </netTcpBinding>
+    </bindings>
+    <client>
+      <endpoint 
+        address="net.tcp://localhost:9090/Telemetry"
+        binding="netTcpBinding"
+        bindingConfiguration="netTCPConfig"
+        behaviorConfiguration="telemetry"
+        contract="Examples.Wcf.IStatusServiceContract"
+        name="StatusService_Tcp" />
+    </client>
+  </system.serviceModel>
+</configuration>
+```
+
+Example project available in
+[examples/wcf/client-netframework](../../examples/wcf/client-netframework/Program.cs)
+folder.
+
+## WCF Client Configuration (.NET Core)
+
+Add the `IClientMessageInspector` instrumentation as an endpoint behavior on the
+clients you want to instrument:
+
+```csharp
+StatusServiceClient client = new StatusServiceClient(binding, remoteAddress);
+client.Endpoint.EndpointBehaviors.Add(new TelemetryEndpointBehavior());
+```
+
+Example project available in
+[examples/wcf/client-core](../../examples/wcf/client-core/Program.cs) folder.
+
+## WCF Server Configuration (.NET Framework)
+
+Add the `IDispatchMessageInspector` instrumentation via a behavior extension on
+the services you want to instrument:
+
+```xml
+<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+  <system.serviceModel>
+    <extensions>
+      <behaviorExtensions>
+        <add 
```

Comment:
@eddynaka @cijothomas I'm getting lint failures on these long lines in the XML examples. Any ideas to get around that?

Reply:
we could disable this rule by creating a config.

something like:
```json
{
  "default": true,
  "MD013": { "line-length": false },
}
```

Reply:
Took a bit of digging but I figured out you can add special comments in the file to suppress warnings: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/38/commits/d07c9c30641aa4745448f17b8b8b6f76ac15aa66